### PR TITLE
 Address feedback on timeRemaining method.

### PR DESCRIPTION
**Pull request by rmcilroy**
Clarifies the internal layout of IdleDeadline with an internal slot holding
the absolute deadline. Clarifies the algorithm in timeRemaining to use this
internal slot and use the underlying primitive for performance.now().

Addresses #54, #55 and #56.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -405,18 +409,20 @@
...
       Security" section of [[!HR-TIME]]. This value is calculated by performing
       the following steps:</p>
       <ol>
-        <li>Let <var>now</var> be the value returned by
-        `[performance.now()]`.</li>
-        <li>Let <var>timeRemaining</var> be `deadline` argument passed to
-        <a>IdleDeadline</a> - <var>now</var>.
-        </li>
+        <li>Let <var>now</var> be a [DOMHighResTimeStamp] representing
+        the current time in milliseconds.</li>
```
Comment (igrigorik) [inline index.html]: Should links this to https://w3c.github.io/hr-time/#dfn-current-high-resolution-time

**Review by igrigorik — COMMENTED** [review]
On a first pass, this looks good, modulo one reference comment. That said, I'd love to get @annevk to sanity check before we land this.

**Comment by igrigorik** [comment]
@annevk can you please take a pass over this?

File: index.html [hunk 2]
```diff
@@ -328,6 +328,10 @@
       <li>An <dfn>idle callback identifier</dfn>, which is a number which MUST
       initially be zero.</li>
     </ul>
+    <p>Each <a>IdleDeadline</a> has an [internal slot] which holds a
+    [DOMHighResTimeStamp] representing the absolute time in milliseconds of the
+    deadline. This slot MUST be populated when the <a>IdleDeadline</a> is
+    created.</p>
```
Comment (annevk) [inline index.html]: You should give the internal slot a name. Something like "Each IdleDeadLine object has a [time] internal slot ..."

File: index.html [hunk 3]
```diff
@@ -599,9 +604,9 @@
 [task source]: http://www.w3.org/TR/2011/WD-html5-20110525/webappapis.html#task-source
 [event loop]: http://www.w3.org/TR/html5/webappapis.html#event-loop
 [DOMHighResTimestamp]: http://www.w3.org/TR/hr-time-2/#dom-domhighrestimestamp
-[performance.now()]: http://www.w3.org/TR/hr-time-2/#widl-Performance-now-DOMHighResTimeStamp
 [pv-hidden]: http://www.w3.org/TR/page-visibility/#dom-document-hidden
 [fully active]: http://www.w3.org/TR/html5/browsers.html#fully-active
 [list of active timers]: http://www.w3.org/TR/html5/webappapis.html#list-of-active-timers
 [requestAnimationFrame]: http://www.w3.org/TR/animation-timing/#dom-windowanimationtiming-requestanimationframe
 [report the error]: http://www.w3.org/TR/html5/webappapis.html#report-the-error
+[internal slot]: https://tc39.github.io/ecma262/#sec-object-internal-methods-and-internal-slots
```
Comment (annevk) [inline index.html]: You probably don't want to reference this. Most other standards also don't say "internal slot" at the moment. They just say that the object has an associated X (where X is a name).

**Review by annevk — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (rmcilroy) [reply index.html]: Done.

File: index.html (hunk 2 above)
Comment (rmcilroy) [reply index.html]: Done.

File: index.html (hunk 3 above)
Comment (rmcilroy) [reply index.html]: Done.

**Comment by rmcilroy** [comment]
Thanks for the reviews. Comments addressed, PTAL.

**Review by annevk — APPROVED** [review]
Thanks, this looks better. (I'd use an Oxford comma personally.)

**Comment by igrigorik** [comment]
Looks great, thanks guys! Merging.

**Pull request merged**